async-nats/src/jetstream/context.rs

```diff
@@ -0,0 +1,56 @@
+use crate::Client;
+use bytes::Bytes;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use serde_json;
+
+/// An error description returned from a jetstream request.
+#[derive(Debug, Deserialize)]
+pub struct Error {
+    /// Code
+    pub code: u64,
```

Comment:
this is supossed to be enum in the future?

Reply:
Probably remain as a number for raw protocol representation. (this is scoped as async_nats::jetstream::context::Error so while its public, you have to dig to get to it).

For one it's bit problematic to encode an enum from an unknown source of unknown value at this low level call site.

For our actual error handling was thinking a bit more along the lines of how std does it, with for example ParseIntError.

```rust
enum CreateStreamErrorKind {
  AlreadyExists,
  MissingName,
  Other,
}

struct CreateStreamError {}
impl CreateStreamError {
  fn kind() -> CreateStreamErrorKind {}
}

impl Context {
  pub async fn create_stream(stream_config: StreamConfig) -> Result<Stream, CreateStreamError> {}
}
```